### ISSUE TITLE
Windows / WAMP - Bad URL - For use incorrect installation

### ISSUE THREAD

**Issue by cavo789** [linuxserver/Heimdall#256]
Hi all

I'm  probably missing something / doing something wrong.

I've installed Heimdall by download the ZIP into a folder on my Windows hard disk, then fired a `composer update` for getting dependencies and newer versions of everything.

`php artisan serve` is working and the site is up and running. So, until now, everything is great.

Now, the problem:

I've created an alias on my WAMP like I always does.
I can access to Heimdall by going to http://127.0.0.1:8080/homepage and it works except for CSS / JS. As you can see below, files are searched under http://127.0.0.1:8080/ (localhost root) and not http://127.0.0.1:8080/homepage (the website root URL).

![2018-10-11_00h17_04](https://user-images.githubusercontent.com/355999/46769142-16ff2180-cceb-11e8-9a15-88bdcc71cd1a.png)

I've tried `APP_URL=http://127.0.0.1:8080/homepage/` in the `.env` file but no, not working.
Also tried `RewriteBase /homepage` in `/public/.htaccess` but here too, not working.

Can you guide me please?

Thanks!

**Comment by KodeStar** [comment]
Yeah, I don't think that is going to work, usually what you do is create a virtual host and point the root to the public dir, you could try navigating to http://127.0.0.1:8080/homepage/public but I'm not sure that will work either, never tried it.

**Comment by cavo789** [comment]
Thanks for the answer. My `homepage` alias is already pointing to the public folder so `homepage/public` will means `/public/public` so not working ;-)

At this time, I don't understand why I need to create a virtual host (I always get copy of PHP pages and create an alias in my Apache then access to the alias). So, here, I'll need to study why I should do like this for Heimdall.

Thank you for your answer here above.

**Comment by KodeStar** [comment]
It was just a guess, I've never tried to do what you are doing.  Heimdall is built on Laravel, so it might be worth you seeing if there are any Laravel generic solutions to your particular setup.

**Comment by KodeStar** [comment]
ok, it looks like the js files are linked to /js/app.js etc, so you would probably need to add a base href, I'll look at changing it to asset() though so it should work with the APP_URL set, might not help though, I think the css should be showing as it's using a mix() call

**Comment by KodeStar** [comment]
Replace your current /resources/views/app.blade.php with the one on master and see if that helps

**Comment by cavo789** [comment]
Much better, thank you! No more problems with .css and .js but well for images and webfonts.

![2018-10-12_21h18_09](https://user-images.githubusercontent.com/355999/46889574-6ae14600-ce64-11e8-8e7a-fb0e18624b37.png)

Thank you for your valuable help!

**Comment by KodeStar** [comment]
Ok, let me see if I can fix that as well, at least I know we are on the right track now

**Comment by KodeStar** [comment]
Ok, try replacing /public/css/app.css and /resources/views/item.blade.php with ones from master and let me know how you get on.

**Comment by cavo789** [comment]
Just copied the two files and bingo!, it's works perfectly for the layout, no more errors in my console.

But, unfortunately, the actions are not using the correct URL.
Only the first action button (with the two arrows, left and right) is using the correct URL (i.e. http://127.0.0.1:8080/homepage/).

![2018-10-14_19h33_27](https://user-images.githubusercontent.com/355999/46920025-4a41f900-cfe8-11e8-840a-5b22d18b35c3.png)

**Comment by KodeStar** [comment]
Well we seem to be getting closer at least!

Have you still got APP_URL set to `APP_URL=http://127.0.0.1:8080/homepage/` in the .env file and `RewriteBase /homepage` in /public/.htaccess?

**Comment by cavo789** [comment]
Yes on both questions; have just tried on a second computer; same result.

```.env
APP_URL=http://localhost:8080/homepage/
```

```.htaccess
<IfModule mod_rewrite.c>
    (unchanged)
</IfModule>

RewriteBase /homepage
```

**Comment by KodeStar** [comment]
try replacing `/app/Providers/AppServiceProvider.php` APP_URL is usually only used for CLI requests, but I've added a check to see if it's been changed from the default `http://localhost` to force the root url.  Whether it will work or not I'm not sure, thanks your your patience being a guinea pig.

**Comment by cavo789** [comment]
Thank you for your tremendous support!

Unfortunally, no, it's not working. With your last change, I loose my CSS/JS.

I've make a diff and have seen that you've added

```php
if(env('APP_URL') != 'http://localhost') {
    \URL::forceRootUrl(env('APP_URL'));
}
```

I've then tried and replace your three lines by this one:

```php
\URL::forceRootUrl('http://127.0.0.1:8080/homepage');
```
I retrieve CSS/JS but the link for actions still doesn't mention the alias so the problem remains: the interface is working but action like creating a new entry, tags, ... uses `http://127.0.0.1:8080/items/create` instead of `http://127.0.0.1:8080/homepage/items/create`.

**Comment by KodeStar** [comment]
Ok, this might work - https://stackoverflow.com/questions/41879110/using-forcerooturl-breaks-some-routing I'll have a go when I get home

**Comment by KodeStar** [comment]
should be fixed in 2.2.1
